tests/test_utils.py

```diff
@@ -289,16 +290,32 @@ def test_memory_profiling():
 
     weights_memory_in_bytes = 128 * 1024 * 1024 * 4 # 512 MiB
 
+    def measure_current_non_torch():
+        free, total = torch.cuda.mem_get_info()
+        current_used = total - free
+        current_torch = torch.cuda.memory_reserved()
+        current_non_torch = current_used - current_torch
+        return current_non_torch
+
     with memory_profiling(baseline_memory_in_bytes=baseline_memory_in_bytes,
-    weights_memory_in_bytes=weights_memory_in_bytes) as result:
+    weights_memory_in_bytes=weights_memory_in_bytes) as result, \
+        monitor(measure_current_non_torch) as monitored_values:
         # make a memory spike, 1 GiB
         spike = torch.randn(256, 1024, 1024, device='cuda', dtype=torch.float32)
         del spike
 
         # Add some extra non-torch memory 256 MiB (simulate NCCL)
         handle2 = lib.cudaMalloc(256 * 1024 * 1024)
 
+    # this is an analytic value, it is exact,
+    # we only have 256 MiB non-torch memory increase
+    measured_diff =monitored_values.values[-1] - monitored_values.values[0]
+    assert measured_diff == 256 * 1024 * 1024
```

Comment:
This is really cool!

I'm a bit confused on how it's useful though, since we're testing a test utility that isn't used in the actual memory profiling? Did we want to enable `monitor(measure_current_non_torch)` during the actual profile run to try to get an accurate measure of the peak non-torch memory usage?

Reply:
> I'm a bit confused on how it's useful though

for the current memory profiling, we mainly use `torch.cuda.memory_reserved()` to replace `torch.cuda.memory_stats()["allocated_bytes.all.current"]` .

the utility function is more about future-proof, we can get the ground-truth non-torch memory, which will help us profile which part of memory can be offloaded in RLHF workload.